airbyte-webapp/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,234 @@
+import classNames from "classnames";
+import { Line } from "rc-progress";
+import { useState } from "react";
+import { useIntl } from "react-intl";
+
+import { getJobStatus } from "components/JobItem/JobItem";
+
+import { AttemptRead, JobConfigType, SynchronousJobRead } from "core/request/AirbyteClient";
+import Status from "core/statuses";
+import { JobsWithJobs } from "pages/ConnectionPage/pages/ConnectionItemPage/JobsList";
+import { formatBytes } from "utils/numberHelper";
+
+import styles from "./ProgressBar.module.scss";
+
+function isJobsWithJobs(job: JobsWithJobs | SynchronousJobRead): job is JobsWithJobs {
+  return (job as JobsWithJobs).attempts !== undefined;
+}
+
+const buttonUnStyle = {
+  background: "none",
+  color: "inherit",
+  border: "none",
+  padding: 0,
+  font: "inherit",
+  cursor: "pointer",
+  outline: "inherit",
+  textDecoration: "underline",
+};
+
+export const ProgressBar = ({
+  job,
+  jobConfigType,
+}: {
+  job: JobsWithJobs | SynchronousJobRead;
+  jobConfigType: JobConfigType;
+}) => {
+  const { formatMessage, formatNumber } = useIntl();
+  const [showStreams, setShowStreams] = useState(false);
+
+  if (jobConfigType !== "sync") {
+    return null;
+  }
+
+  let numeratorRecords = -1;
+  let denominatorRecords = -1;
+  let totalPercentRecords = -1;
+  let numeratorBytes = -1;
+  let denominatorBytes = -1;
+  let elapsedTimeMS = -1;
+  let timeRemaining = -1;
+  let timeRemainingString = "";
+  const unEstimatedStreams: string[] = [];
+  let displayProgressBar = true;
+  let latestAttempt: AttemptRead | undefined;
+
+  const jobStatus = getJobStatus(job);
+
+  // colors from `_colors.scss` TODO: Use the SCSS variables maybe?
```

Comment:
Can we import these colors from SCSS somehow or reference classnames in the global scope?